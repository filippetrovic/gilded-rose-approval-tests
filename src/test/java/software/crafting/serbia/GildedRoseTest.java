package software.crafting.serbia;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

public class GildedRoseTest {

  @Test
  public void approvalTestFixtureForGildedRose() {
    // Given
    final String[] names = {
        "name"
    };

    // Then
    CombinationApprovals.verifyAllCombinations(this::doStuff, names);

  }

  private String doStuff(String name) {
    Item[] items = new Item[]{
        new Item(name, 0, 0)
    };

    final GildedRose gildedRose = new GildedRose(items);

    gildedRose.updateQuality();

    return items[0].toString();
  }
}
