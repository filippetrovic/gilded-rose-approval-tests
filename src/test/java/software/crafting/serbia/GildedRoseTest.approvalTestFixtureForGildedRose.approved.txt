[name] => name, -1, 0 
